frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -169,7 +173,38 @@ function BookInfo:show(file, book_props, metadata_updated_caller_callback)
     UIManager:show(self.kvp_widget)
 end
 
-function BookInfo:getBookProps(file, book_props, no_open_document)
+-- Returns customized metadata.
+function BookInfo.customizeProps(original_props) -- , filepath)
+    local custom_props = {} -- stub
+    original_props = original_props or {}
+
+    local props = {}
+    for _i, prop_key in ipairs(BookInfo.props) do
+        props[prop_key] = custom_props[prop_key] or original_props[prop_key]
+    end
+    return props
+end
+
+-- Returns document metadata (opened document or book (file) metadata or custom metadata).
+function BookInfo.getDocProps(ui, file, book_props, no_open_document, no_customize)
+    local original_props, filepath, display_title
+    if ui then -- currently opened document
+        original_props = ui.doc_settings:readSetting("doc_props")
+        filepath = ui.document.file
```

Comment:
(Dunno if you should use or check `file` when both `ui` and `file` are provided - it feels that when additional arguments are provided, they tweak the behaviour of the previous ones, so if `ui` (generic in its meaning) and `file` (precise) both provided, I would expect file to be used - and `if file == ui.document.file`, use the info from `ui`.)

Reply:
> when both `ui` and `file` are provided

For now I cannot imagine this case.